compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -997,25 +1010,31 @@ class ProtobufGenerator(
                 s"__fieldsMap.getOrElse(__fields.get(${field.getIndex}), $t).asInstanceOf[$baseTypeName]"
               }
 
-            transform(field).apply(e, enclosingType = field.enclosingType)
-        }
-        val oneOfs = message.getOneofs.asScala.map { oneOf =>
-          val elems = oneOf.fields.map { field =>
-            val typeName =
-              if (field.isEnum) "_root_.com.google.protobuf.Descriptors.EnumValueDescriptor"
-              else field.baseSingleScalaTypeName
-            val e =
-              s"__fieldsMap.get(__fields.get(${field.getIndex})).asInstanceOf[_root_.scala.Option[$typeName]]"
-            (transform(field) andThen FunctionApplication(field.oneOfTypeName)).apply(
-              e,
-              EnclosingType.ScalaOption
-            )
-          }
-          val expr =
-            elems.reduceLeft((acc, field) => s"$acc\n    .orElse[${oneOf.scalaTypeName}]($field)")
-          s"${oneOf.scalaName.asSymbol} = $expr\n    .getOrElse(${oneOf.empty})"
+            Seq(transform(field).apply(e, enclosingType = field.enclosingType))
```

Comment:
Why does the order here matter?

Reply:
Are we https://github.com/scalapb/ScalaPB/blob/c3e44a0\/scalapb-runtime/non-jvm/src/main/scala/com/google/protobuf/wrappers/Int32Value.scala#L89? If I remember correctly, I updated that because of a compilation error when running e2e. But indeed, instead of cascading the order changes in the constructor, we could use named parameters for internal code invoking it though.

Reply:
I had a closer look, and only the `oneofs` parameters were explicitely named. I am generalizing usage of named parameters for calling the constructor to avoid that.